tiledb/sm/array_schema/dimension.h

```diff
@@ -239,12 +248,73 @@ class Dimension {
          << domain[0] << ", " << domain[1] << "] on dimension '" << dim->name()
          << "'";
       *err_msg = ss.str();
+
       return false;
     }
 
     return true;
   }
 
+  /**
+   * Performs correctness checks on the input range. Returns `true`
+   * upon error and stores an error message to `err_msg`.
+   *
+   * Applicable to integral domains.
```

Comment:
I know the parameters are obvious from context, but let's document the parameters here and below. I'm mostly concerned that we document that `range` is mutated as a side-effect.

same below.